app/screens/AppNavigator/screens/Settings/components/RowLanguageItem.tsx

```diff
@@ -0,0 +1,73 @@
+import { LanguagePersistence } from '@api'
+import { ThemedIcon, ThemedText, ThemedTouchableOpacity } from '@components/themed'
+import { WalletAlert } from '@components/WalletAlert'
+import { useLanguageContext } from '@contexts/LanguageProvider'
+import { EnvironmentLanguage } from '@environment'
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { tailwind } from '@tailwind'
+import { getLocaleByLanguageName, translate } from '@translations'
+
+import * as React from 'react'
+import { useCallback } from 'react'
+import { SettingsParamList } from '../SettingsNavigator'
+
+export function RowLanguageItem (props: { language: EnvironmentLanguage }): JSX.Element {
+  const navigation = useNavigation<NavigationProp<SettingsParamList>>()
+  const { language, setLanguage } = useLanguageContext()
+  const onPress = useCallback(async () => {
```

Comment:
don't use useCallback here anymore. It's too buggy on mobile (Like poolswap error). Let's use it only on pure functions